pxtsim/runtime.ts

```diff
@@ -245,21 +246,30 @@ namespace pxsim {
                         this.lock = false;
                     }
                 })
+            });
         }
 
-        get handler() {
-            return this._handler;
+        get handlers() {
+            return this._handlers;
         }
 
-        set handler(a: RefAction) {
-            if (this._handler) {
-                pxtcore.decr(this._handler);
-            }
+        addHandler(a: RefAction) {
+            this._handlers.push(a);
+            pxtcore.incr(a)
+        }
 
-            this._handler = a;
+        setHandler(a: RefAction) {
+            this._handlers.forEach(old => pxtcore.decr(old))
```

Comment:
This seems to remove all old handlers, not only the foreground one

Reply:
Good catch.

Reply:
but setHandler only called for foreground.

Reply:
Oh, I think I misunderstood.